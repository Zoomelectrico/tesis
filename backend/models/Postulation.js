const mongoose = require("mongoose");

const postulationSchema = new mongoose.Schema({
  electoralGroup: {
    required: "Debe proveer un grupo electoral",
    type: mongoose.Schema.Types.ObjectId,
    ref: "ElectoralGroup"
  },
  passed: {
    type: Number, // Yes: 1 | No: 0
    default: 0
  },
  fce: [{ name: String, dni: Number, school: String, charge: String }],
  sports: [{ name: String, dni: Number, school: String }],
  culture: [{ name: String, dni: Number, school: String }],
  services: [{ name: String, dni: Number, school: String }],
  academic: [{ name: String, dni: Number, school: String }],
  responsibility: [{ name: String, dni: Number, school: String }],
  academicCouncil: {
    name: String,
    email: String,
    dni: Number,
    phone: String,
    school: String
  },
  facultyCouncil: [
    {
      name: String,
      dni: Number,
      school: String,
      faculty: String,
      substitute: Number,
      facultyKey: String
    }
  ],
  schools: [
    {
      name: String,
      dni: Number,
      charge: String,
      school: String,
      schoolKey: String
    }
  ],
  schoolCouncil: [
    {
      name: String,
      dni: Number,
      school: String,
      substitute: Number,
      schoolKey: String
    }
  ]
});

module.exports = mongoose.model("Postulation", postulationSchema);
