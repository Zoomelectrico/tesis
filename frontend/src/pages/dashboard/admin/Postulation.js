import React, { useState, useEffect } from "react";
import { Container, Col, Row, Card, Button } from "reactstrap";
import { Header, notify, Loading } from "../../../components";
import { get, post } from "../../../utils";
import { Checkout } from "../postulate";

const Postulation = ({ location, history }) => {
  const params = new URLSearchParams(location.search);
  const [state, setState] = useState({
    loading: true,
    postulation: {},
    demand: {}
  });
  useEffect(() => {
    const fetch = async () => {
      try {
        const data = await get(`demand/${params.get("id")}`);
        console.log(data);
        setState({
          loading: false,
          postulation: data.demand.postulation,
          demand: data.demand
        });
      } catch (err) {
        notify("Ha Ocurrido un Error. Intente Refrescando el navegador", false);
      }
    };
    fetch();
  }, []);

  const accept = async e => {
    try {
      e.preventDefault();
      const { pid, id } = e.target.dataset;
      const data = await post("demand-accept-pos", { pid, id });
      console.log(data);
      history.push(
        "/app/dashboard/demands?reason=Postulacion-registrada-correctamente&bool=true"
      );
    } catch (err) {
      console.log(err);
      notify("Ha ocurrido un Error al Guardar");
    }
  };

  return (
    <>
      <Header />
      {state.loading ? (
        <Container fluid className="mt--7">
          <Row>
            <Col sm="12">
              <Card>
                <Loading />
              </Card>
            </Col>
          </Row>
        </Container>
      ) : (
        <>
          <Checkout postulation={state.postulation} />
          <Container>
            <Row>
              <Col sm="12" className="d-flex justify-content-end">
                <Button
                  color="success"
                  className="my-auto"
                  onClick={accept}
                  data-id={state.demand._id}
                  data-pid={state.postulation._id}
                >
                  Aceptar Postulacion
                </Button>
              </Col>
            </Row>
          </Container>
        </>
      )}
    </>
  );
};

export default Postulation;
