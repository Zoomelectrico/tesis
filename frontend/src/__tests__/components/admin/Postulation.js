import React from 'react';
import { shallow } from 'enzyme';
import Postulation from '../../../pages/dashboard/admin/Postulation';

describe('Electoral Groups', () => {
  it('Render Correctly', () => {
    const wrapper = shallow(<Postulation />);
    expect(wrapper).toMatchSnapshot();
  });
});
