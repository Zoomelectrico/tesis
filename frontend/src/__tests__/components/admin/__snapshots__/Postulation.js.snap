// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Electoral Groups Render Correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Postulation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <Container
          className="mt--7"
          fluid={true}
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Card
                tag="div"
              >
                <Loading />
              </Card>
            </Col>
          </Row>
        </Container>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Row
            tag="div"
          >
            <Col
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Card
                tag="div"
              >
                <Loading />
              </Card>
            </Col>
          </Row>,
          "className": "mt--7",
          "fluid": true,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Card
                tag="div"
              >
                <Loading />
              </Card>
            </Col>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Card
                tag="div"
              >
                <Loading />
              </Card>,
              "sm": "12",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Loading />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <Container
            className="mt--7"
            fluid={true}
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Card
                  tag="div"
                >
                  <Loading />
                </Card>
              </Col>
            </Row>
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Row
              tag="div"
            >
              <Col
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Card
                  tag="div"
                >
                  <Loading />
                </Card>
              </Col>
            </Row>,
            "className": "mt--7",
            "fluid": true,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Col
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Card
                  tag="div"
                >
                  <Loading />
                </Card>
              </Col>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Card
                  tag="div"
                >
                  <Loading />
                </Card>,
                "sm": "12",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Loading />,
                  "tag": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
